web/src/components/graph/areagraph.js

```diff
@@ -0,0 +1,132 @@
+import React, {
+  useState,
+  useEffect,
+  useRef,
+  useMemo,
+} from 'react';
+import { scaleTime, scaleLinear } from 'd3-scale';
+import { first, last } from 'lodash';
+import moment from 'moment';
+
+import AreaGraphGradients from './areagraphgradients';
+import AreaGraphLayers from './areagraphlayers';
+import GraphBackground from './graphbackground';
+import GraphHoverLine from './graphhoverline';
+import ValueAxis from './valueaxis';
+import TimeAxis from './timeaxis';
+
+const X_AXIS_HEIGHT = 20;
+const Y_AXIS_WIDTH = 35;
+const Y_AXIS_PADDING = 4;
+
+const getTimeScale = (containerWidth, datetimes, currentTime) => scaleTime()
+  .domain([first(datetimes), currentTime ? moment(currentTime).toDate() : last(datetimes)])
+  .range([0, containerWidth]);
+
+const getValueScale = (containerHeight, maxTotalValue) => scaleLinear()
+  .domain([0, maxTotalValue * 1.1])
+  .range([containerHeight, Y_AXIS_PADDING]);
+
+const AreaGraph = React.memo(({
+  id,
+  layers,
+  currentTime,
+  maxTotalValue,
```

Comment:
can this be calculated / memoized?

Reply:
You're right, it was very easy to infer this info from the `layers` props :) See https://github.com/tmrowco/electricitymap-contrib/pull/2171/commits/cbb72f14063b532a73d0819d6ec973a89c26492a.

Reply:
By the way, similarly it was easy to infer `datetimes` from the `layers` prop so I did it in https://github.com/tmrowco/electricitymap-contrib/pull/2171/commits/231759db2ba696ec053ac384f672e2aeb305bf23.